src/text_parser.py

```diff
@@ -0,0 +1,203 @@
+import re
+import nltk
+from nltk.corpus import words
+from nltk.stem import WordNetLemmatizer
+#import mistune
+import markdown
+'''preprocess data for input'''
+
+# remove previously referenced comments
+#remove every line that starts with ">"
+def remove_reference(text):
+    pattern = r'''^>.*'''
+    result = re.sub(pattern, "", text, 0, re.MULTILINE | re.IGNORECASE | re.VERBOSE)
+    return result
+
+#Note: does not work well with short senteces.
+#Note: does not work well with short senteces.
+def contain_non_english(text):
+    try:
+        return detect(text) != "en"
+    except:
+        return True  #if can't detect return true
+
+# an edge case to deal with
+'''
+```yml
+    staticDir:
+      - "static"
+      - "../dist"
+    ```
+'''
+def remove_markdown(text):
+    pattern = r"""
+        ^(?:\ {4}.+\n)+(?!)         # Ignore content of all indented text
+        |(^```(?:[^`]+|`(?!``))*```)  # Do not stop at single ` backticks within ```...``` blocks
+        |(```.+?```) # remove inline ```...```
+        """
+    #why doesn't work if we remove ^
+    result = re.sub(pattern, "MarkdownCode", text, 0, re.MULTILINE | re.IGNORECASE | re.VERBOSE)
+    return result
+
+def transform_markdown(text):
+    return markdown.markdown(text)#mistune.markdown(text)
+
+#replace inline highlighted code with text
+def remove_inline_code(text):
+    pattern = r'''(`.+?`)'''
+    result = re.sub(pattern, "InlineCode", text, 0, re.MULTILINE | re.IGNORECASE | re.VERBOSE)
+    return result
+
+def remove_empty_space(text):
+    pass
+
+def sub_PlusOne(text):
+    pattern = r"""(\+1)"""
+    result = re.sub(pattern,"plus one", text,0,re.MULTILINE | re.IGNORECASE | re.VERBOSE)
+    return result
+
+#remove @
+def replace_mention(text):
+    pattern = r'''@[A-Za-z0-9-]+''' # remove @
+    result = re.sub(pattern, "@friend", text, 0, re.MULTILINE | re.IGNORECASE | re.VERBOSE)
+    return result
+
+
+def remove_url(text):
+    pattern = r'''https?://[^ ]+'''
+    result = re.sub(pattern, "", text, 0, re.MULTILINE | re.IGNORECASE | re.VERBOSE)
+    return result
+
+def remove_newline(text):
+    #text = text.encode('utf-8').replace('\n', ' ').replace('\r', '').replace('\t','')
+    text = text.replace('\n', ' ').replace('\r', '').replace('\t', '')
+    return text
+
+# just remove all :
+def remove_emoji_marker(text):
+
+    pattern = r''':'''
+    text = re.sub(pattern, "", text, 0, re.MULTILINE | re.IGNORECASE | re.VERBOSE)
+    return text
+
+#has probelms. Deal with this later. Removing letters that are not supposed to be removed
```

Comment:
seems contradictory

Reply:
This function is not used anywhere else.. I've deleted it now..